### ISSUE TITLE
laas-mode overrides org-mode aas

### ISSUE THREAD

**Issue by mohkale** [tecosaur/LaTeX-auto-activating-snippets#15]
I'm trying to bind a snippet to use in org-mode, while also using this package laas-mode.

```lisp
(use-package aas
  :straight t)

(use-package laas
  :straight t
  :hook ((org-mode . laas-mode)
         (LaTeX-mode . laas-mode))
  :config
  (put 'aas-set-snippets 'lisp-indent-function 'defun)
  (aas-set-snippets 'laas-mode
    :cond #'laas-mathp
    "On"    "\\mathcal{O}(n)"
    "O1"    "\\mathcal{O}(1)"
    "Olog"  "\\mathcal{O}(\\log n)"
    "Olon"  "\\mathcal{O}(n \\log n)"
    "<<" nil)

  (aas-set-snippets 'org-mode
    "<<" (lambda! ()
           (yas-expand-snippet
            (concat ",#+BEGIN_SRC $1\n"
                    "$0\n"
                    ",#+END_SRC")))))
```

But nothing I do seems to work :?. I know laas-mode has it's own binding for `<<` so I disabled it, but the org-mode snippet still checks `laas-mathp` despite me not specifiying it as a condition for `aas-set-snippets`.

**Comment by ymarco** [comment]
You forgot to activate the keymap you created for org-mode.

Your configuration only activates laas-mode, which takes care of calling aas-mode and activating the laas keymap. Your << snippet is defined in aas's org-mode keymap

Does it work with something like this?

  (use-package laas
    :straight t
    :hook ((org-mode . laas-mode)
+          (org-mode . aas-activate-for-major-mode)
           (LaTeX-mode . laas-mode))
    :config ...)

**Comment by mohkale** [comment]
@ymarco 

Yes, you're correct. Thank you.

I'm still getting an error with my binding at the start of the buffer though:

```
Debugger entered--Lisp error: (wrong-type-argument number-or-marker-p nil)
  laas--no-backslash-before-point\?()
  aas-expand-snippet-maybe("<<" (closure (t) nil (interactive) nil (yas-expand-snippet (concat "#+BEGIN_SRC $1\n" "$0\n" "#+END_SRC"))) nil)
  #f(compiled-function () #<bytecode 0x207588126261>)()
  aas-post-self-insert-hook()
  org-self-insert-command(1)
  funcall-interactively(org-self-insert-command 1)
  call-interactively(org-self-insert-command nil nil)
  command-execute(org-self-insert-command)
```

I think it's because `(char-before)` is nil in `laas--no-backslash-before-point?`. 

```diff
--- /home/mohkale/.config/dotfiles/prog/editors/emacs/lisp/straight/repos/LaTeX-auto-activating-snippets/laas.el
+++ #<buffer laas.el>
@@ -384,7 +384,8 @@
 
 (defun laas--no-backslash-before-point? ()
   "Check that the char before `point' is not a backslash."
-  (/= (char-before) ?\\))
+  (when-let ((ch (char-before)))
+    (/= ch ?\\)))
 
 
 (apply #'aas-set-snippets 'laas-mode laas-basic-snippets)
```

**Comment by mohkale** [comment]
Actually if I apply the above patch the snippet doesn't expand at all at the start of the line, even if there's no longer any error triggered.

**Comment by ymarco** [comment]
The patch is incorrect: returning nil when char-before is nil there amounts to
saying that there *is* a backslash before point and preventing snippet expansion

**Comment by mohkale** [comment]
@ymarco 

But I'd like snippets (this specific snippet) to expand at the start of the line? Is there a reason `laas--no-backslash-before-point?` was put into `aas-global-condition-hook` instead of included in `laas-mathp`? I don't imagine it's good for laas to interfere with org-mode snippets defined outside of laas.

**Comment by ymarco** [comment]
I'm not sure I want to move it away from the global hook.
It's supposed to stop the case of e.g "I wrote the backslash of \sin myself and the sin snippet expanded so now I have \\sin". (This really happened, and was when I added this hook).

It definitely shouldn't be put in laas-mathp, I don't want functions like these to have surprises.

Your patch just incorrect. I'll push a better one, this edge case of laas--no-backslash-before-point? has needed fixing for a while.
